dashboard/app/models/user_level.rb

```diff
@@ -24,6 +26,10 @@ class UserLevel < ActiveRecord::Base
   belongs_to :level
   belongs_to :script
 
+  scope :attempted, -> { where.not(best_result: nil) }
+  scope :passed, -> { where('best_result >= ?', ActivityConstants::MINIMUM_PASS_RESULT) }
+  scope :perfected, -> { where('best_result > ?', ActivityConstants::MAXIMUM_NONOPTIMAL_RESULT) }
```

Comment:
Yes, the intent is to include all "optimal" scores, including free-play scores and the like.